### ISSUE TITLE
False positive: `__()` rejects `Closure` values in `$replace`

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1442]
**Describe the bug**

`__()` rejects a `Closure` value in `$replace`, which Laravel supports at runtime.

The stub at `stubs/common/Foundation/helpers.phpstub:361` declares:

```
@param  array<string, scalar|null>  $replace
```

`Illuminate\Translation\Translator::makeReplacements()` handles a `Closure` value explicitly:

```php
foreach ($replace as $key => $value) {
    if ($value instanceof Closure) {
        $line = preg_replace_callback(
            '/<'.$key.'>(.*?)<\/'.$key.'>/',
            fn ($args) => $value($args[1]),
            $line
        );

        continue;
    }
    // ...
}
```

The correct parameter type is therefore:

```
@param  array<string, (\Closure(string): string)|scalar|null>  $replace
```

`\Closure`, not `callable`: the branch tests `instanceof Closure`, so a callable string or array falls through to the `is_object()` / string cast path instead and never receives the callback treatment.

**Code to reproduce**

```php
$ok = __('greeting', ['name' => 'Alies']);

// false positive on the line below
$bad = __('greeting', ['tag' => static fn (string $s): string => '<b>' . $s . '</b>']);
```

Reported today:

```
InvalidArgument: Argument 2 of __ expects array<string, null|scalar>, but array{tag: pure-Closure(string):non-falsy-string} provided
```

**Scope**

* `trans()` and `trans_choice()` are not affected. Neither carries a stub (both are routed through handlers), so callers keep Laravel's own untyped `@param array $replace`.
* Only the parameter is wrong. The return type stays as it is: `($key is null ? null : string|array)` matches Laravel's `($key is null ? null : array|string)`.
* The `Closure` branch is present on every supported Laravel version (verified at `v11.55.1`, `v12.14.0`, `v12.67.0`, `v13.3.0`, `v13.26.1`), so the fix belongs in `stubs/common/` with no version gate.
* The stub line is byte identical on `master` and `3.x`, so both branches need it.

**Origin**

The narrow `array<string, scalar|null>` shape was adopted from Larastan in #1151. Larastan widened it to include the closure form in `v3.11.0`. Our copy went stale.

**Impacted Versions**

```text
laravel/framework 13.26.0
vimeo/psalm       7.0.0-beta19
```

**Additional context**

A regression type test is warranted: the false positive is reproducible and the parameter is a conditional-adjacent shape that a plain stub sync could silently re-narrow.
